### PR TITLE
protodoc: Integrate artefact data

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/protodoc/BUILD [hunk 1]
```diff
@@ -20,24 +21,71 @@ envoy_proto_library(
...
+    cmd = """$(location :manifest_to_json) $(location //docs:protodoc_manifest.yaml) $@""",
+    tools = [
+        ":manifest_to_json",
         "//docs:protodoc_manifest.yaml",
-        "//docs:v2_mapping.json",
+    ],
+)
+
+envoy_genjson(
+    name = "data_srcs",
```
Comment (phlax) [inline tools/protodoc/BUILD:48]: @RyanTheOptimist see here (the filter needs a bit of cleaning up)

**Comment by phlax** [comment]
@htuch i think this one should land next it optimizes a little but more just makes the algo/data derivation more bazel-explicit

**Comment by htuch** [comment]
I'm not sure if this is clearer TBH. Can you add some comments and explain what is going on? We're replacing some fairly readable Python with a complex JSON filter expression inside a BUILD file.

**Comment by phlax** [comment]
comments added - apologies for force push, my local branch was already rebased

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**